src/Razor/test/Microsoft.VisualStudio.LanguageServer.ContainedLanguage.Test/TestVirtualDocumentFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage.Test;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServer.ContainedLanguage
+{
+    internal class TestVirtualDocumentFactory : VirtualDocumentFactoryBase
+    {
+        public const string HostDocumentContentTypeNameConst = "TestHostContentTypeName";
+        public const string LanguageContentTypeNameConst = "TestLanguageContentTypeName";
+        public const string LanguageFileNameSuffixConst = "__virtual.test";
+
+        public static IContentType LanguageLSPContentTypeInstance { get; } = new TestContentType(LanguageContentTypeNameConst);
+
+        public TestVirtualDocumentFactory(
+            IContentTypeRegistryService contentTypeRegistryService,
+            ITextBufferFactoryService textBufferFactoryService,
+            ITextDocumentFactoryService textDocumentFactoryService,
+            FileUriProvider fileUriProvider
+            ) : base(contentTypeRegistryService, textBufferFactoryService, textDocumentFactoryService, fileUriProvider) { }
+
+        protected override IContentType LanguageLSPContentType => LanguageLSPContentTypeInstance;
+
```

Comment:
Should also add language buffer properties here to validate that they're set as expected in the successful TryGet test

Reply:
Agreed, will do.

---
In reply to: [452551402](https://github.com/dotnet/aspnetcore-tooling/pull/2201#discussion_r452551402) [](ancestors = 452551402)